pkg/server/test/list_objects.go

```diff
@@ -308,55 +334,40 @@ func TestListObjectsRespectsMaxResults(t *testing.T, ds storage.OpenFGADatastore
 // Used to avoid compiler optimizations (see https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go)
 var listObjectsResponse *commands.ListObjectsResponse //nolint
 
-func BenchmarkListObjectsWithReverseExpand(b *testing.B, ds storage.OpenFGADatastore) {
+func BenchmarkListObjectsWithMaxConcurrentReadsEqualToOne(b *testing.B, ds storage.OpenFGADatastore) {
 	ctx := context.Background()
 	store := ulid.Make().String()
 
-	model := &openfgav1.AuthorizationModel{
-		Id:            ulid.Make().String(),
-		SchemaVersion: typesystem.SchemaVersion1_1,
-		TypeDefinitions: []*openfgav1.TypeDefinition{
-			{
-				Type: "user",
-			},
-			{
-				Type: "document",
-				Relations: map[string]*openfgav1.Userset{
-					"viewer": typesystem.This(),
-				},
-				Metadata: &openfgav1.Metadata{
-					Relations: map[string]*openfgav1.RelationMetadata{
-						"viewer": {
-							DirectlyRelatedUserTypes: []*openfgav1.RelationReference{
-								typesystem.DirectRelationReference("user", ""),
-							},
-						},
-					},
-				},
-			},
-		},
-	}
-	err := ds.WriteAuthorizationModel(ctx, store, model)
-	require.NoError(b, err)
-
-	n := 0
-	for i := 0; i < 100; i++ {
-		var tuples []*openfgav1.TupleKey
+	model, modelID, numberObjectsAccesible := setupListObjectsBenchmark(b, ds, store)
 
-		for j := 0; j < ds.MaxTuplesPerWrite(); j++ {
-			obj := fmt.Sprintf("document:%s", strconv.Itoa(n))
-			user := fmt.Sprintf("user:%s", strconv.Itoa(n))
+	listObjectsQuery := commands.NewListObjectsQuery(ds, commands.WithMaxConcurrentReads(1))
 
-			tuples = append(tuples, tuple.NewTupleKey(obj, "viewer", user))
+	var r *commands.ListObjectsResponse
 
-			n += 1
-		}
+	ctx = typesystem.ContextWithTypesystem(ctx, typesystem.New(model))
 
-		err = ds.Write(ctx, store, nil, tuples)
-		require.NoError(b, err)
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		r, _ = listObjectsQuery.Execute(ctx, &openfgav1.ListObjectsRequest{
+			StoreId:              store,
+			AuthorizationModelId: modelID,
+			Type:                 "document",
+			Relation:             "viewer",
+			User:                 "user:maria",
+		})
+		require.LessOrEqual(b, len(r.Objects), numberObjectsAccesible)
 	}
 
-	listObjectsQuery := commands.NewListObjectsQuery(ds)
+	listObjectsResponse = r
+}
+
+func BenchmarkListObjectsWithMaxConcurrentReadsEqualToThirty(b *testing.B, ds storage.OpenFGADatastore) {
+	ctx := context.Background()
+	store := ulid.Make().String()
+
+	model, modelID, numberObjectsAccesible := setupListObjectsBenchmark(b, ds, store)
+
+	listObjectsQuery := commands.NewListObjectsQuery(ds, commands.WithMaxConcurrentReads(30))
```

Comment:
For these tests (BenchmarkListObjectsWithMaxConcurrentReadsEqualToOne, BenchmarkListObjectsWithMaxConcurrentReadsEqualToThirty, BenchmarkListObjectsWithMaxResultsEqualToZero, etc.), other than setting commands.WithMaxConcurrentReads different, are there any difference?  If not, can we have common function to make things easier to read?

Reply:
will do